tests/link/jax/test_random.py

```diff
@@ -79,8 +82,34 @@ def test_RandomStream():
     srng = RandomStream(seed=123)
     out = srng.normal() - srng.normal()
 
-    fn = function([], out, mode=jax_mode)
+    with pytest.warns(
+        UserWarning,
+        match=r"The RandomType SharedVariables \[.+\] will not be used",
+    ):
+        fn = function([], out, mode=jax_mode)
     jax_res_1 = fn()
     jax_res_2 = fn()
 
-    assert np.array_equal(jax_res_1, jax_res_2)
+    assert not np.array_equal(jax_res_1, jax_res_2)
+
+
+@pytest.mark.parametrize("rng_ctor", (np.random.RandomState, np.random.default_rng))
+def test_random_updates(rng_ctor):
+    original_value = rng_ctor()
```

Comment:
Perhaps we should set a seed value here to guarantee the consistency of the `assert` that follows?

Reply:
The f() != f()? That should be incredibly rare to happen by chance.

Reply:
No chance is better than any in this scenario, especially given the effort required to avoid it entirely.

Reply:
Done